internal/controller/management/upgrade/serviceaccounts_test.go

```diff
@@ -0,0 +1,137 @@
+package upgrade
+
+import (
+	"context"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+
+	rbacapi "github.com/akuity/kargo/api/rbac/v1alpha1"
+)
+
+func TestHasOldAnnotations(t *testing.T) {
+	tests := []struct {
+		name        string
+		annotations map[string]string
+		expected    bool
+	}{
+		{
+			name:        "No old annotations",
+			annotations: map[string]string{"some-key": "some-value"},
+			expected:    false,
+		},
+		{
+			name:        "Has old sub annotation",
+			annotations: map[string]string{OldAnnotationKeySub: "some-value"},
+			expected:    true,
+		},
+		{
+			name:        "Has old email annotation",
+			annotations: map[string]string{OldAnnotationKeyEmail: "some-value"},
+			expected:    true,
+		},
+		{
+			name:        "Has old groups annotation",
+			annotations: map[string]string{OldAnnotationKeyGroups: "some-value"},
+			expected:    true,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			result := hasOldAnnotations(tt.annotations)
+			assert.Equal(t, tt.expected, result)
+		})
+	}
+}
+
+func TestServiceAccountReconciler_Reconcile(t *testing.T) {
+	scheme := runtime.NewScheme()
+	_ = corev1.AddToScheme(scheme)
```

Comment:
It's a good idea in our unit tests to test assumptions as well as outcomes. An failure in test setup could affect the outcome of the test cases. If the setup fails silently, we end up with a harder time tracking down the root cause.

```suggestion
	err := corev1.AddToScheme(scheme)
	require.NoError(t, err)
```